R/dm.R

```diff
@@ -270,29 +269,24 @@ as_dm.default <- function(x) {
   # Automatic name repair
   names(x) <- vctrs::vec_as_names(names2(x), repair = "unique")
 
-  src <- tbl_src(x[[1]])
-
-  # FIXME: Check if all sources identical
+  # Check if all sources are identical
+  if (!all_same_source(x)) abort_not_same_src()
 
   # Empty tibbles as proxy, we don't need to know the columns
   # and we don't have keys yet
   proxies <- map(x, ~ tibble(a = 0))
   data_model <- datamodelr::dm_from_data_frames(proxies)
 
-  new_dm(src, x, data_model)
+  new_dm(x, data_model)
 }
 
 tbl_src <- function(x) {
   if (is.data.frame(x)) {
-    src_df(env = new_environment(x))
+    NULL
```

Comment:
`dplyr::src_df(env = .GlobalEnv)` -- can you extract a function for the default local source?